language-antlr-utils/src/main/java/de/jplag/antlr/AbstractAntlrLanguage.java

```diff
@@ -22,8 +25,25 @@ protected AbstractAntlrLanguage(AbstractAntlrParserAdapter<?> parser) {
         this.parser = parser;
     }
 
+    /**
+     * New instance, without pre initialized parser. If you use this constructor, you need to override the initializeParser
+     * method.
+     */
+    protected AbstractAntlrLanguage() {
+        this.parser = null;
+    }
+
     @Override
     public List<Token> parse(Set<File> files) throws ParsingException {
+        if (this.parser == null) {
+            this.parser = this.initializeParser();
+        }
+
         return this.parser.parse(files);
     }
+
+    protected AbstractAntlrParserAdapter<?> initializeParser() {
```

Comment:
Maybe there shoudl also be a doc comment here about it needing to be implemented when using the empty constructor and it not being needed when using the other one